__tests__/02.patch-state.ava.ts

```diff
@@ -52,79 +52,86 @@ if (Workspace.networkIsSandbox()) {
     ],
   ]);
 
-  workspace.test('View state', async (test, {contract, ali}) => {
-    await ali.call(contract, 'set_status', {message: 'hello'});
+  // eslint plugin bug, this is not inside an async that cannot use await
+  // eslint-disable-next-line promise/prefer-await-to-then
```

Comment:
Instead of using a top-level promise chain like this, we should make this example leverage an async `before()` handler that dispatches and awaits for the promise, then places the resulting data into the `context` object for the rest of the suite to reference, which is idiomatic AVA usage.

Reply:
Sounds good. I'm doing this in another branch, which intend to drop workspaces-ava and workspaces-init. As you can see top-level tests are using workspaces-ava, so in that branch, I replace workspaces-ava with raw ava + workspaces-js. Some finished work:

```
import {Workspace} from 'near-workspaces';
import anyTest, {TestFn} from 'ava';

const test = anyTest as TestFn<{workspace: Workspace}>;
test.before(t => {
  t.context.workspace = Workspace.init(async ({root}) => ({
    contract: await root.createAndDeploy(
      'status-message',
      '__tests__/build/debug/status_message.wasm',
    ),
    ali: await root.createAccount('ali'),
  }));
});

test('Root gets null status', async t => {
  await t.context.workspace.fork(async ({contract, root}) => {
    const result: null = await contract.view('get_status', {
      account_id: root,
    });
    t.is(result, null);
  })
});
...
```